packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts

```diff
@@ -119,11 +120,21 @@ function getDividedStyleSheetsDependingOnItsPurpose( rules: Array<Rule> ) {
 		editingViewStylesContent += objectWithDividedStyles.editingViewStyles;
 
 		if ( objectWithDividedStyles.rootDefinitions.length ) {
-			rootDefinitionsList.push( ...objectWithDividedStyles.rootDefinitions );
+			objectWithDividedStyles.rootDefinitions.forEach( definitionText => {
+				const listOfDefinitions = definitionText.split( '\n' );
+
+				listOfDefinitions.forEach( definition => {
+					if ( definition && rootDefinitionsSet.has( definition ) ) {
+						showInfoOfDuplicatedVariableDefinition( definition );
```

Comment:
Variable defined later should override previous value (the same as in CSS).

Reply:
TBH whole method should be modified. I choose `Set` to quickly eliminate the duplicates, but each `rootDefinitionsSet` helds a string like `--variable: red;` and it compares it with same one not just a variable name.

To make it work as expected `Set` should be changed into `Map` and store values as key/value: `'--variable'` => `'red'`. Check if the key already exists and if yes, replace just the value (and can make some warning that value `X` of variable `FOO` was changed to `Y` ).